addons/test_discuss_full/tests/test_performance.py

```diff
@@ -99,7 +99,7 @@ def test_init_messaging(self):
         self.maxDiff = None
         self.users[0].flush()
         self.users[0].invalidate_cache()
-        with self.assertQueryCount(emp=98):
+        with self.assertQueryCount(emp=101):
```

Comment:
Just to make sure, what are the extra queries in this case?
I would guess it's the queries to fetch the access rules and apply them?

Reply:
I would do the "clean" in the fetch of star (rarely done rpc) rather than in the counter (always done rpc), so this should be revertable.

Reply:
Maybe the clean could even be called only from the JS if we detect an empty message is present on the result?

Reply:
I would still lean towards doing it on the python side, don't you think?
At least, I think it would be a bit weird to do this cleaning on the JS side, but maybe I'm wrong, maybe you find it cleaner this way :smile:

Reply:
We'll see, as suggested by PM I think we should keep this PR focused on fixing the bug to merge is as soon as possible, and create a new PR where we think about cleaning them if necessary.